packages/tinacms/src/react.tsx

```diff
@@ -37,3 +37,15 @@ export function useTina<T extends object>(props: {
   }, [])
   return { data } as any
 }
+
+export function useEditState(): { edit: boolean } {
+  const [edit, setEdit] = React.useState(false)
+  if (typeof window !== 'undefined') {
+    window.addEventListener('message', (event) => {
+      if (event.data?.type === 'tina:updateData') {
```

Comment:
This will only trigger when the user is using a `useTina` hook elsewhere on the page. but if they're on a page without the `useTina` hook they could still be in an edit state (ie. in the iframe). 

For some context, some of our users are using the `useEditState` hook to modify the image URL (when in edit mode, keep the assets.tina.io portion of the URL, when not in edit mode, use the relative path.

I think the fix here can be to add very similar logic to the current postMessage stuff, but to put it in the `preview.tsx` file and yeah, probably using a separate `type`. (I think adding the `tina:updateData` type is good and should stay.

Reply:
@jeffsee55 I updated the PR to resolve this issue.